packages/workbox-sw/WorkboxSW.mjs

```diff
@@ -0,0 +1,166 @@
+/*
+  Copyright 2017 Google Inc.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+
+import './_version.mjs';
+
+/**
+ * This module is a single import that can be used to dynamically import
+ * additional Workbox modules with no effort.
+ *
+ * @module workbox-sw
+ */
+
+const CDN_PATH = `WORKBOX_CDN_ROOT_URL`;
+
+// TODO Make this list be generated during build time using the package.json's.
+// workbox namespace value.
+const MODULE_KEY_TO_NAME_MAPPING = {
+  backgroundSync: 'background-sync',
+  core: 'core',
+  expiration: 'cache-expiration',
+  googleAnalytics: 'google-analytics',
+  strategies: 'runtime-caching',
+  precaching: 'precaching',
+  routing: 'routing',
+};
+
+/**
+ * This class can be used to make it easy to use the various parts of
+ * Workbox.
+ */
+class WorkboxSW {
+  /**
+   * You can specify whether to treat as a debug
+   * build and whether to use a CDN or a specific path when importing other
+   * workbox-modules
+   */
+  constructor() {
+    this.v = {};
+    this._options = {
+      debug: self.location.hostname === 'localhost',
+      modulePathPrefix: null,
+      modulePathCb: null,
+    };
+
+    this._env = this._options.debug ? 'dev' : 'prod';
+    this._modulesLoaded = false;
+
+    return new Proxy(this, {
+      get(target, key) {
+        if (target[key]) {
+          return target[key];
+        }
+
+        const moduleName = MODULE_KEY_TO_NAME_MAPPING[key];
+        if (moduleName) {
+          target._loadModule(`workbox-${moduleName}`);
+        }
+
+        return target[key];
+      },
+    });
+  }
+
+  /**
+   * Updates the configuration options.
+   *
+   * @param {Object=} [options]
+   */
+  config(options = {}) {
```

Comment:
`config()` feels vague to me, would prefer `setConfig()` or `configure()` just so it has an action associated with it.

Reply:
Agreed, changed to `setConfig()`.